test/Language/Cimple/PrettySpec.hs

```diff
@@ -0,0 +1,101 @@
+module Language.Cimple.PrettySpec where
+
+import           Test.Hspec                   (Spec, describe, it, shouldBe)
+
+import qualified Data.Text                    as Text
+import           Language.Cimple.IO           (parseText)
+import           Language.Cimple.Pretty       (ppTranslationUnit)
+import           Text.PrettyPrint.ANSI.Leijen (displayS, renderCompact)
+
+getRight :: Either String a -> a
+getRight (Left  err) = error err
+getRight (Right ok ) = ok
+
+pretty :: String -> IO String
+pretty = fmap (show . ppTranslationUnit . getRight) . parseText . Text.pack
+
+compact :: String -> IO String
+compact =
+    fmap (flip displayS "" . renderCompact . ppTranslationUnit . getRight)
+        . parseText
+        . Text.pack
+
+
+spec :: Spec
+spec = do
+    describe "renderPretty" $ do
+        it "pretty-prints a simple C function" $ do
+            pp <- pretty "int a(void) { return 3; }"
+            pp `shouldBe` unlines
+                [ "int a(void)"
+                , "{"
+                , "  return 3;"
+                , "}"
+                ]
+
+        it "pretty-prints a typedef struct" $ do
+            pp <- pretty "typedef struct Foo { int32_t x; } Foo;"
+            pp `shouldBe` unlines
+                [ "typedef struct Foo {"
+                , "  int32_t x;"
+                , "} Foo;"
+                ]
+
+    describe "renderCompact" $ do
+        it "pretty-prints a simple C function" $ do
+            pp <- compact "int a(void) { return 3; }"
+            pp `shouldBe` unlines
+                [ "int a(void)"
+                , "{"
+                , "return 3;"
+                , "}"
+                ]
+
+        it "pretty-prints a typedef struct" $ do
+            pp <- compact "typedef struct Foo { int32_t x; } Foo;"
+            pp `shouldBe` unlines
+                [ "typedef struct Foo {"
+                , "int32_t x;"
+                , "} Foo;"
+                ]
+
+        it "respects comment styles" $ do
+            compact "/* foo bar */" >>= (`shouldBe` "/* foo bar\n */\n")
+            compact "/** foo bar */" >>= (`shouldBe` "/** foo bar\n */\n")
+            compact "/*** foo bar */" >>= (`shouldBe` "/*** foo bar\n */\n")
+            compact "/**** foo bar */" >>= (`shouldBe` "/*** foo bar\n */\n")
+
+        it "supports punctuation in comments" $ do
+            compact "/* foo.bar,baz-blep */"
+                >>= (`shouldBe` "/* foo. bar, baz - blep\n */\n")
+            compact "/* foo? */" >>= (`shouldBe` "/* foo ?\n */\n")
+
+        it "formats number ranges and negative numbers without spacing" $ do
+            compact "/* 123 - 456 */" >>= (`shouldBe` "/* 123-456\n */\n")
+            compact "/* - 3 */" >>= (`shouldBe` "/* -3\n */\n")
+            compact "/* a-b */" >>= (`shouldBe` "/* a - b\n */\n")
+
+        it "formats pointer types with east-const" $ do
```

Comment:
East const best const?

Reply:
It's mostly just easier to pretty-print, but yes :).